common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java

```diff
@@ -223,6 +223,47 @@ && verifySignature(authenticator.packageName)) {
         return null;
     }
 
+    /**
+     * Determines which app is the broker based on having the work account registration in Account Manager.
+     *
+     * Known issue: When we're in an AccountManager callback (Especially on older Android devices, i.e. Android 10)
+     * Android For Work throws a SecurityException when we're calling AccountManager.getAuthenticatorTypes()
+     * i.e. E.g. Company Portal main process can call this freely, broker process can call this freely, but once running in AccountManager,
+     * on the work profile (user != 0), apparently sometimes it tries to get accounts from user 0 (personal profile) and fails.
+     *
+     * In such case, this method will return null.
+     *
+     * @return PackageName of the broker
+     */
+    @NonNull
+    // TODO: Consolidate getCurrentActiveBrokerPackageName and getValidActiveBrokerPackageName into one.
+    public String getValidActiveBrokerPackageName() throws ClientException {
+        final String methodTag = TAG + ":getValidActiveBrokerPackageName";
+
+        String authenticatorPackageName = null;
+        try {
+            final AuthenticatorDescription[] authenticators = AccountManager.get(mContext).getAuthenticatorTypes();
+            Logger.info(methodTag, authenticators.length + " Authenticators registered.");
+            for (AuthenticatorDescription authenticator : authenticators) {
+                if (authenticator.type.equals(BROKER_ACCOUNT_TYPE)) {
+                    authenticatorPackageName = authenticator.packageName;
```

Comment:
You could also invoke `verifySignatureAndThrow` here and then just return early from the loop

Reply:
If you do that, then you can probably entirely get rid of the `authenticatorPackageName` variable